region/maridia/inner-green/Shaktool Room.json

```diff
@@ -189,38 +199,126 @@
+      ]
+    },
+    {
+      "link": [2, 1],
+      "name": "Shaktool Room Reverse",
+      "notable": true,
+      "requires": [
+        "h_canUsePowerBombs"
+      ],
+      "clearsObstacles": ["A", "B"],
+      "setsFlags": ["f_ShaktoolDoneDigging"],
+      "note": "Let the Snails (Yards) dig through the sand and follow them. They will not dig while off screen or while Samus is facing them, even while morphed."
+    },
+    {
+      "link": [2, 1],
+      "name": "G-Mode Morph Reverse Shaktool",
+      "notable": true,
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": true
+        }
+      },
+      "requires": [
+        {"or": [
+          {"and": [
+            "Gravity",
+            "h_canArtificialMorphIBJ"
+          ]},
+          "h_canArtificialMorphSpringBall"
+        ]},
+        "h_canArtificialMorphPowerBomb"
+      ],
+      "clearsObstacles": ["A", "B"],
+      "setsFlags": ["f_ShaktoolDoneDigging"],
+      "flashSuitChecked": true,
+      "note": [
+        "The Snails will dig through the sand without showing any visual difference. Each block they pass through gets closer to overloading PLMs.",
+        "In order to PB the left wall before PLMs are overloaded, it is necessary to only allow one snail to dig, either by bombing one or by facing one to prevent movement until it is off screen.",
+        "If using mobile g-mode, it is easiest to get past the bottom one and prevent it from digging. If using immobile g-mode, it is easiest to have whichever one hits Samus go off screen.",
+        "In the morph tunnel, it is important to place the PB as far right as possible, while still hitting the PB blocks, otherwise the Snails will have overloaded the PLMs and the blocks won't break.",
+        "The green stone changes one tile earlier than the PB wall, and can be a good visual cue to place the PB.",
+        "After using the PB, exit the sand, leave G-Mode, and wait for Shaktool. Note that the Snails will not dig while off screen or while Samus is facing them, even while she is morphed."
+      ]
+    },
+    {
+      "link": [2, 1],
+      "name": "Grapple Teleport (Vanilla)",
       "entranceCondition": {
         "comeInWithGrappleTeleport": {
           "blockPositions": [[5, 3], [7, 2]]
         }
       },
-      "requires": []
+      "requires": [
+        "h_ShaktoolVanillaFlag"
+      ],
+      "devNote": [
+        "In the vanilla game, grapple teleporting here does not set the flag 'f_ShaktoolDoneDigging', as Samus does not spend a frame on the right side of the room."
+      ]
     },
     {
       "link": [2, 1],
-      "name": "Grapple Teleport Door Lock Skip",
+      "name": "Grapple Teleport (Symmetric)",
+      "entranceCondition": {
+        "comeInWithGrappleTeleport": {
+          "blockPositions": [[5, 3], [7, 2]]
+        }
+      },
+      "requires": [
+        "h_ShaktoolSymmetricFlag"
+      ],
+      "setsFlags": ["f_ShaktoolDoneDigging"]
```

Comment:
could use a note here and on all the similar strats. the name of the helper is not intuitive.